dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/ingest/tabulardata/impl/plugins/csv/CSVFileReader.java

```diff
@@ -110,14 +111,14 @@ private void init() throws IOException {
     @Override
     public TabularDataIngest read(BufferedInputStream stream, File dataFile) throws IOException {
         init();
-
+        Charset charset = selectCharset();
         if (stream == null) {
             throw new IngestException(IngestError.UNKNOWN_ERROR);
         }
         TabularDataIngest ingesteddata = new TabularDataIngest();
         DataTable dataTable = new DataTable();
 
-        BufferedReader localBufferedReader = new BufferedReader(new InputStreamReader(stream));
+        BufferedReader localBufferedReader = new BufferedReader(new InputStreamReader(stream, charset));
```

Comment:
Can we add some test that will check if we have selected charset correctly? We could add a simple csv file with some encoding like `win-1250` and check if we produced correct data file or that generated variables doesn't contain weird characters